app/models.py

```diff
@@ -354,22 +358,37 @@ class Config:
 class UpdateSession(BaseModel):
     """Model for the body of the request that updates a session"""
 
-    has_quiz_ended: bool
+    has_quiz_ended_first_time: bool
+    has_quiz_started_first_time: bool
 
     class Config:
         schema_extra = {
             "example": {
-                "has_quiz_ended": True,
+                "has_quiz_ended_first_time": False,
+                "has_quiz_started_first_time": True,
             }
         }
 
 
+class UpdateSessionResponse(BaseModel):
+    """Model for the response of request that updates a session"""
+
+    time_remaining: Optional[int]
+
+    class Config:
+        schema_extra = {"example": {"time_remaining": 300}}
+
+
 class SessionResponse(Session):
     """Model for the response of any request that returns a session"""
 
     is_first: bool
+    has_quiz_started: Optional[bool] = False
     has_quiz_ended: Optional[bool] = False
     session_answers: List[SessionAnswer]
+    quiz_start_resume_time: Optional[datetime]
```

Comment:
why `start` and `resume` in same variable name? Little weird

Reply:
because we can either start session or resume session when quiz opens. i'm capturing the moment when the start/resume button is clicked

Reply:
^this will change with the improved implementation. But the basic funda still remains valid. The events and their names should be independent!